### ISSUE TITLE
How do we include on-disk repositories using lorax?

### ISSUE THREAD

**Issue by abotelho-cbn** [weldr/lorax#1200]
Hi all,
The mkksiso tool has an option "-a ADD_PATHS, --add ADD_PATHS" which allows someone to add directories on the host into the ISO, which then enables one to effectively include an on-disk repository that a Kickstart could then go and fetch.

But let's say I wanted to do something like that with Lorax. How can I approach this? This is to generate a custom boot.iso which contains extra packages present on the disk, for fully offline installations. BaseOS would have to be included, and then an "extra" repo directory, which would be an assortment of packages with createrepo_c applied to it.

When doing my testing when running lorax, generated ISOs have no repos by default to begin with.

Cheers

**Comment by VladimirSlavik** [comment]
tl;dr: `lorax (...) -s file://$REPO_DIR/`

---

For a real example, take a look at how anaconda tests build boot.iso on demand. It's a bit more convoluted, but it runs every day, here you go...

First we build packages:
https://github.com/rhinstaller/anaconda/blob/master/.github/workflows/kickstart-tests.yml#L161

Then we mount that into the container:
https://github.com/rhinstaller/anaconda/blob/master/.github/workflows/kickstart-tests.yml#L180

In the container, we `createrepo_c` that dir (arguably a bit late):
https://github.com/rhinstaller/anaconda/blob/master/dockerfile/anaconda-iso-creator/lorax-build#L24

Finally lorax runs with that:
https://github.com/rhinstaller/anaconda/blob/master/dockerfile/anaconda-iso-creator/lorax-build#L30

**Comment by bcl** [comment]
@VladimirSlavik @abotelho-cbn is asking about repos on the built iso, not repos used to build the iso :)

This kind of thing is what pungi (https://pagure.io/pungi) is used for, it makes sure all the required packages are present, and creates the dvd with the repositories included. You could hack something together by modifying the lorax templates, but you are better off using pungi to make sure everything is in the format expected by Anaconda.
